web/modules/custom/ihc_social/src/AuthProviderFactory.php

```
@@ -13,27 +13,33 @@
 class AuthProviderFactory {
 
   /**
-   * Constructs a new AuthProviderFactory object.
+   * Returns a new Social Provider object depending on the type requested
    *
    * @return AbstractProvider;
    */
   public static function create($type, $classPath) {
     $className = __NAMESPACE__ . $classPath;
 
     try {
+      $key_value_storage = \Drupal::keyValue('ihc_social.' . $type);
+      if ( ! $key_value_storage->has('client_id') && ! $key_value_storage->has('client_secret')) {
```

This should probably throw an exception if either of these are unavailable, not strictly if both are unavailable